test/uk/gov/hmrc/gform/auth/AuthServiceSpec.scala

```diff
@@ -248,6 +254,33 @@ class AuthServiceSpec extends ExampleData with SpecWithFakeApp {
     result.futureValue should be(AuthRedirect(""))
   }
 
+  it should "not authorise an Ofsted user when they have not been successfully authenticated by the AWS ALB" in {
+    val result =
+      authService
+        .authenticateAndAuthorise(
+          formTemplateAWSALBMissingHTTPHeader,
+          lang,
+          requestUri,
+          getAffinityGroup,
+          ggAuthorisedSuccessful)
+    result.futureValue should be(AuthBlocked("You are not authorized to access this service"))
+  }
+
+  it should "authorise an Ofsted user when they have been successfully authenticated by the AWS ALB" in {
+    val jwt =
+      "eyJ0eXAiOiJKV1QiLCJraWQiOiI5MTgyZjVlZS0xMTBiLTQ3NWItOWUzNC02OTcyZjdjMTFhYjQiLCJhbGciOiJFUzI1NiIsImlzcyI6Imh0dHBzOi8vY29nbml0by1pZHAuZXUtd2VzdC0yLmFtYXpvbmF3cy5jb20vZXUtd2VzdC0yX0dpSDBTWTBqOSIsImNsaWVudCI6InZjdWZ1Zm05YjdvaTVzazJkdW1mZXJnbG8iLCJzaWduZXIiOiJhcm46YXdzOmVsYXN0aWNsb2FkYmFsYW5jaW5nOmV1LXdlc3QtMjo0MjAyMDgyNTUwODg6bG9hZGJhbGFuY2VyL2FwcC9hbGItZ2Zvcm1zLXFhLW9mc3RlZC8xZWRiY2FjNmQxNjVjYzkyIiwiZXhwIjoxNTU4MDg1ODQxfQ==.eyJzdWIiOiIyMGU5YjI0My03NDcxLTQwODEtYmUxZS1mY2I1ZGEzM2ZkNWEiLCJlbWFpbF92ZXJpZmllZCI6InRydWUiLCJlbWFpbCI6Im1pa2FpbC5raGFuQGRpZ2l0YWwuaG1yYy5nb3YudWsiLCJ1c2VybmFtZSI6IjIwZTliMjQzLTc0NzEtNDA4MS1iZTFlLWZjYjVkYTMzZmQ1YSIsImV4cCI6MTU1ODA4NTg0MSwiaXNzIjoiaHR0cHM6Ly9jb2duaXRvLWlkcC5ldS13ZXN0LTIuYW1hem9uYXdzLmNvbS9ldS13ZXN0LTJfR2lIMFNZMGo5In0=.7mBYP9FKoIPf7JBZ9qMcm90n1UNICrWyCBadi5xcjs0pKGGWFlDZLVKmvHSmpCK731JCrz49VPwDQYfsY_I_UA=="
```

Comment:
Would it be possible (for future reference) to describe how was this token created?

Reply:
The token comes from the ALB. You can generate dummy ones for tests.

Reply:
Would it be possible to put link how to do that into the code?

Reply:
You just need to read up on JWT - it's not complicated.

Reply:
Is adding link complicated?